File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int cents;

    //Get amount of dollars from user
    do
    {
        dollars = get_float("Please enter amount of dollars.");
        cents = round(dollars * 100);
    }
    while (dollars < 0);

    int count = 0;
    int change = 0;

    //Use quarters, while possible
    while ((change + 25) <= cents)
    {
        change = change + 25;
        count ++;
    }

    //Use dimes, while possible
    while ((change + 10) <= cents)
    {
        change = change + 10;
        count ++;
    }

    //Use nickels, while possible
    while ((change + 5) <= cents)
    {
        change = change + 5;
        count ++;
    }

    //Use pennies, while possible
    while ((change + 1) <= cents)
    {
        change = change + 1;
        count ++;
    }

    //Print minimum amount of coins
    printf("%i\n", count);
}

File: pset6/cash/cash.py
from cs50 import get_float

# Get amount of dollars from user
while True:
    dollars = get_float("Change owed: ")
    cents = round(dollars * 100)

if dollars > 0:
    break

counter = 0
change = 0

# Use quarters, while possible
while change + 25 <= cents:
    change = change + 25
    counter += 1

# Use dimes, while possible
while change + 10 <= cents:
    change = change + 10
    counter += 1

# Use nickels, while possible
while change + 5 <= cents:
    change = change + 5
    counter += 1

# Use pennies, while possible
while change + 1 <= cents:
    change = change + 1
    counter += 1

# Print minimum amount of coins
print(counter)
